### DISCUSSION TITLE
Little help please with Tuya-DALI bridge external converter

### DISCUSSION THREAD

**Discussion by Gelisob** [Koenkk/zigbee2mqtt#21073] [Devices]
Heya, trying to follow the how-to's to get one zigbee-to-dali bridge working with Z2M. But it's my first time and i'm not sure is my case more complicated.

Before details - my problem is that the external converter seems to be ignored, maybe, and the device keeps getting recognized as the zigbee rgbw light thingy.

The real thing i have:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/22873920/d24127fc-553e-4660-8a08-43e289ed6a9d)

When first paired - by default it gets recognized as "Zigbee RGBW light" - my guess is, because in the belly there is chip from some zigbee dimmer device and it has DALI bridge things "bolted" on. 

Here's what Z2M translates it into:
![image](https://github.com/Koenkk/zigbee2mqtt/assets/22873920/73ba7b9c-51b6-47ba-9a15-48d3ae4aa7b4)

following the external converters guide, I was attempting to force it into being a custom new thing so that I could start poking it for it's unmapped endpoints, i guess thats the term.

So i made TS0504B_DALI.js converter file from example and put it into \\homeassistant.local\share\zigbee2mqtt\external_converters folder.
in z2m HA addon configuration I added

```
external_converters:
  - TS0504B_DALI.js

```

And here's the insides of my beginning converter:

```
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const extend = require('zigbee-herdsman-converters/lib/extend');
const e = exposes.presets;
const ea = exposes.access;
const tuya = require('zigbee-herdsman-converters/lib/tuya');

const definition = {
    fingerprint: [
        {
            modelID: 'TS0504B',
            manufacturerName: '_TZB210_5jn6an2y',
        },
    ],
    model: 'TS0504B_DALI',
    vendor: 'CENFAYA',
    description: 'Zigbee & RF DALI master, 6 in 1',
    fromZigbee: [tuya.fz.datapoints /*, Custom DALI converters here */],
    toZigbee: [tuya.tz.datapoints /*, Custom DALI converters here */],
    onEvent: tuya.onEventSetTime, // Add this if you are getting no converter for 'commandMcuSyncTime'
    configure: tuya.configureMagicPacket,
    exposes: [
        // Define your device's functionalities here
    ],
    meta: {
        tuyaDatapoints: [
            // List specific Tuya datapoints here, if applicable
        ],
    },
};

module.exports = definition;
```

Is there anything obvious sticking out to some of you more knowledgeable peeps?
Or should I approach it differently given that it's not totally unknown device but gets recognized as something already. 
To keep the post little more manageable I'll add a herdsman log snippet I managed to grab, as an attached file.

Thanks to anyone who takes time to look into it, would be nice to get support for this device going, can bring Dali things into fold through zigbee.

[tuya-dali-herdsman-logSnippet03.txt](https://github.com/Koenkk/zigbee2mqtt/files/14073562/tuya-dali-herdsman-logSnippet03.txt)

**Comment by Gelisob** [comment]
the default detected device exposes following menus and none of those do anything to DALI light connected to the devices dali outputs. The device does slight blink it's red LED when sending command, like on or off state. But it does not seem to be forwarded to DALI..
![image](https://github.com/Koenkk/zigbee2mqtt/assets/22873920/c1a1ee8b-e53f-4180-88b6-b728fcd690dc)

**Reply by rafareusch** [reply]
> the default detected device exposes following menus and none of those do anything to DALI light connected to the devices dali outputs. The device does slight blink it's red LED when sending command, like on or off state. But it does not seem to be forwarded to DALI.. ![image](https://private-user-images.githubusercontent.com/22873920/301447092-c1a1ee8b-e53f-4180-88b6-b728fcd690dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk0MDAwNzQsIm5iZiI6MTcyOTM5OTc3NCwicGF0aCI6Ii8yMjg3MzkyMC8zMDE0NDcwOTItYzFhMWVlOGItZTUzZi00MTgwLTg4YjYtYjcyOGZjZDY5MGRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIwVDA0NDkzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTIwMGJlZTg2NTRjMGE0ZDNiZDJmZmM3YTlmOTliNzAzOWI3MTA1OTBhNGNjYWExOGYxMmUyY2E5NjQ5OTFlNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.kmbOkR2G1UC5oobVic4Ax0ekemZkbgLxRDocJ5xIJy8)

Have you figured this out?
I have the same issue on  my end

**Reply by Gelisob** [reply]
I didn't find anyone more knowledgeable in the matter to go deep with this. What responses I got and what I was able to understand, was that I can not do it without information from the manufacturer. I did try to ask for the required ends through seller. They told me that they will ask but I did not hear back from them.
Ended up getting hd67941-b2 for a hundred, someone close by was selling.
Would be nice to get the tuya unit working too ofc.

**Comment by vdhmax** [comment]
I also have these in my new house. I can't even discover them in my Home Assistant instance (with SONOFF dongle) in Zigbee2MQTT... Was anyone able to make these work?

**Comment by vdhmax** [comment]
An update from my side. I managed to connect 4 of these to my Home Assistant setup. Even though I have 4 of the same controllers, they get recognised as 2 different devices. While these devices do have several options to configure them, none actually do anything to my Dali driver/lights … see screenshots. 
![IMG_1076](https://github.com/user-attachments/assets/2eff8ea8-2a5b-4290-8c80-bc33c3272b13)
![IMG_1077](https://github.com/user-attachments/assets/357ba456-2c5c-42a6-8121-0fc98248bdc1)
![IMG_1078](https://github.com/user-attachments/assets/30306c7f-55ea-4436-9a19-8b1ef69fff6c)
![IMG_1079](https://github.com/user-attachments/assets/4b76b638-6542-48ee-98ba-5c8d8f20b6c5)

**Reply by Gelisob** [reply]
correct, you see zigbee side that uses some sort of common led dimmer circuit. And HA uses that to show you the led dimmer that you see from icons. With this kind of pairing, no commands seem to be passed on to dali broadcast.

**Comment by Greeve99** [comment]
You need to set the correct output driver mode by pressing the match/set button. (Switch, dimmer ..etc) The indicator light color on the device will indicate the set mode of the device. 
The
Main issue I have is if the Dali device is controlled from another Dali device the change (eg brightness or on foo state) is not updated on the Zigbee side.

**Reply by Gelisob** [reply]
because yours is also connected to zigbee as a led dimmer device and it does not know anything about this device being a dali device and does not show you any dali hooks in Z2M ?

**Comment by rafareusch** [comment]
Hey, have you ever managed to have a solution?
